patroni/postgresql/config.py

```diff
@@ -154,6 +154,37 @@ def mtime(filename):
         return None
 
 
+class ConfigWriter(object):
+
+    def __init__(self, filename):
+        self._filename = filename
+        self._fd = None
+
+    def __enter__(self):
+        self._fd = open(self._filename, 'w')
+        self.writeline('# Do not edit this file manually!\n# It will be overwritten by Patroni!')
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self._fd:
+            self._fd.close()
+
+    def writeline(self, line):
+        self._fd.write(line)
+        self._fd.write('\n')
```

Comment:
Why not one write with `{line}\n`?

Reply:
Just avoiding useless string concatenation.